### ISSUE TITLE
Add ability to filter by workflows waiting on a timer

### ISSUE THREAD

**Issue by antmendoza** [temporalio/ui#1002]
## Is your feature request related to a problem? Please describe.

It would be nice to have a filter in the UI to filter workflows waiting on a timer

**Comment by karneyli** [comment]
Would definitely be useful!  Also some way of seeing a workflow execution is in an `Await` in the execution detail page in the UI.  

When I have an execution in a `Workflow.await` with a timeout provided I see this in the event history:

<img width="1618" alt="image" src="https://github.com/temporalio/ui/assets/315319/07e82335-596f-4ba5-9ab4-0e66ae381868">

But without a timeout provided, the `Workflow.await` looks like this:

<img width="1360" alt="image" src="https://github.com/temporalio/ui/assets/315319/c200d10e-f535-4088-9d6e-de5fecb7b47b">

In the latter example, there's no discernable indication that it's currently in a `Workflow.await` at all.
